### PR TITLE
feat(storybooks): build storybooks for CI and deploy to mozilla-fxa/storybooks

### PR DESCRIPTION
**Pull request by lmorchard**
- tweaks to CircleCi config to build & deploy storybooks for pull
  requests and master merges

- updates for yarn

- tweaks to use github personal access token

- refactor storybook webpack customizations to use shared function in
  fxa-react, along with fixes to module resolution

- add fxa-content-server as dependency for fxa-react, since it seems to
  reuse content server styles that, in turn, need photon-colors

issue #5385

### CODE REVIEW

**Comment by FxaStorybooks** [comment]
# Storybook Deployment
Pull Request: https://mozilla-fxa.github.io/storybooks/pulls/5429/
Commit: https://mozilla-fxa.github.io/storybooks/commits/e8deb6756f295c3de45e5717ddd6cc5150831d54/

**Comment by lmorchard** [comment]
Okay, I think this is probably ready for review. That comment from @FxaStorybooks is basically the result of what this PR should do for future PRs after merge. Having some trouble with apparently unrelated test failures, but maybe one of these retries will be the charm

**Comment by FxaStorybooks** [comment]
# Storybook Deployment
Pull Request: https://mozilla-fxa.github.io/storybooks/pulls/5429/
Commit: https://mozilla-fxa.github.io/storybooks/commits/e8deb6756f295c3de45e5717ddd6cc5150831d54/

**Comment by FxaStorybooks** [comment]
# Storybook Deployment
Pull Request: https://mozilla-fxa.github.io/storybooks/pulls/5429/
Commit: https://mozilla-fxa.github.io/storybooks/commits/e8deb6756f295c3de45e5717ddd6cc5150831d54/

**Comment by lmorchard** [comment]
Hmm, after seeing this comment repeated for the same commit on test retries, it might make sense to add some code to skip the storybook build if the gh-pages repository already has the current PR commit in it.

File: packages/fxa-react/package.json [hunk 1]
```diff
@@ -23,6 +23,7 @@
     "build-storybook": "build-storybook"
   },
   "dependencies": {
+    "fxa-content-server": "workspace:*",
```
Comment (dannycoates) [inline packages/fxa-react/package.json:26]: what does fxa-react need from content-server?

Comment (dannycoates) [reply packages/fxa-react/package.json:26]: ah, https://github.com/mozilla/fxa/blob/master/packages/fxa-react/components/Survey/index.scss#L1 🙈

File: packages/fxa-react/.storybook/webpack.config.js [hunk 2]
```diff
@@ -1,26 +1 @@
...
-    'css-loader',
-    // Compiles Sass to CSS
-    'sass-loader'
-    ],
-  });
-  config.resolve.extensions.push('.scss', '.css');
-
-  return config;
-};
+module.exports = require("../../fxa-react/storybooks").customizeWebpackConfig;
```
Comment (jodyheavener) [inline packages/fxa-react/.storybook/webpack.config.js:1]: Looks like this is referencing from within fxa-react already, so could you just get away with `../storybooks`?

File: packages/fxa-react/storybooks.js [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+const customizeWebpackConfig = ({ config, mode }) => {
+  // https://storybook.js.org/docs/configurations/typescript-config/#setting-up-typescript-to-work-with-storybook-1
+  config.module.rules.push({
+    test: /\.(ts|tsx)$/,
+    loader: require.resolve("babel-loader"),
+    options: {
+      presets: [["react-app", { flow: false, typescript: true }]],
+    },
+  });
+  config.resolve.extensions.push(".ts", ".tsx");
```
Comment (jodyheavener) [inline packages/fxa-react/storybooks.js:17]: I suspect, with a rebase, this file should see a few prettier changes (notably quotes)

**Comment by lmorchard** [comment]
>  I agree about not posting a new comment for test retries; is that something that can be knocked out here, or should a new issue be filed?

I'd push that to a follow-up issue. There are bound to be lots of other improvements we think of as we try to actually use this thing.

**Comment by lmorchard** [comment]
Rebased, let's see if this passes CI again

**Comment by jodyheavener** [comment]
> I'd push that to a follow-up issue. There are bound to be lots of other improvements we think of as we try to actually use this thing.

Totally. I'm already stoked about this as it is!

**Comment by FxaStorybooks** [comment]
# Storybook Deployment
Pull Request: https://mozilla-fxa.github.io/storybooks/pulls/5429/
Commit: https://mozilla-fxa.github.io/storybooks/commits/a391868ac31dd4433227d48347406c9478ef8bb8/

**Comment by lmorchard** [comment]
@dannycoates  Is this *actually* approved, or do the changes still need to be made?

**Comment by lmorchard** [comment]
Rebased and did some path tweaks, let's see if this passes CI again

File: packages/fxa-react/storybooks.js [hunk 4]
```diff
@@ -0,0 +1,42 @@
+const { resolve } = require('path');
+
+const additionalJSImports = {
```
Comment (lmorchard) [inline packages/fxa-react/storybooks.js:3]: Might be nice at some point to extract this from rescripts.js & storybooks.js into a shared config, but didn't want to get into that right now. Maybe if we add a third path here...

**Comment by FxaStorybooks** [comment]
# Storybook Deployment
Pull Request: https://mozilla-fxa.github.io/storybooks/pulls/5429/
Commit: https://mozilla-fxa.github.io/storybooks/commits/a71aa0cb74da2ef5b0f814e86ccf6438627e9620/

**Comment by lmorchard** [comment]
Okay, I think I made all the right changes, things still work, I'm going to merge it before anything else bitrots

**Pull request merged**